### PR TITLE
modify close before opening logic

### PR DESCRIPTION
**Pull request by iFwu**

### CODE REVIEW

**Comment by ohjames** [comment]
Would it be possible to create a unit test demonstrating the fix?

**Comment by ohjames** [comment]
I can't tell what issue this is supposed to be fixing, let me know some details and I'll consider merging it.

File: src/index.ts [hunk 1]
```diff
@@ -79,6 +79,9 @@ export default function makeWebSocketObservable<T extends WebSocketPayload = Web
     socket.onopen = () => {
       isSocketOpen = true
       observer.next(getWebSocketResponses)
```
Comment (ohjames) [inline src/index.ts]: Okay I get this now, sorry for the confusion. Could you put this `next` in an else branch of the next condition, there's no point issuing that event when the stream is closed.

Comment (iFwu) [reply src/index.ts]: Actually, it won't fire any event since the observer has unsubscribed, but it makes the code clearer

**Pull request merged**